### ISSUE TITLE
Meet W3C WAI's Web Content Accessibility Guidelines 2.0, Level AA conformance

### ISSUE THREAD

**Issue by joshdanielson** [loconomics/loconomics#528]
As a provider of web content to potentially many people with disabilities, Loconomics needs to comply with W3C's WAI's Web Content Accessibility Guidelines 2.0, Level AA conformance.

### Useful links
[How to Meet WCAG 2.0](https://www.w3.org/WAI/WCAG20/quickref/#qr-text-equiv-all)
[Tips for Getting Started with Web Accessibility](https://www.w3.org/WAI/gettingstarted/tips/)
[Web Accessibility in Mind](http://webaim.org/)
[Reference to Section 508 requirements](https://www.access-board.gov/guidelines-and-standards/communications-and-it/about-the-ict-refresh/background/comparison-table-of-wcag2-to-existing-508-standards)
[SSA Accessibility Best Practices Library](https://www.ssa.gov/accessibility/bpl/default.htm)
[Trusted Tester Program](https://www.dhs.gov/sites/default/files/publications/Trusted_Tester_Fact_Sheet_0.pdf)

**Comment by joshdanielson** [comment]
@ndarilek could you review the accessibility /aboutMe activity? I went through all the fields in it this afternoon. I gave them labels, some additional input attributes, and added some fieldset and legend tags. I'd like both your and @IagoSRL 's feedback, and then I'm going to go through the rest of the app inputs and repeat the process. THANK YOU!

**Comment by ndarilek** [comment]
How do I access it? I placed the text "accessibility /aboutMe" after the 
"!" with no luck. Also prepended a /, same result.

**Comment by IagoSRL** [comment]
@ndarilek the activity is just /aboutMe (URL to local is http://localhost:8811/appDebug.html#!/aboutMe).

**Comment by joshdanielson** [comment]
@ndarilek I'm going to revise the legends as I understand the screen readers may read them with each field. So I'm going to try for shorter ones and add some "super" legends if that makes sense

**Comment by ndarilek** [comment]
* Presentation of "Address line 2:" and "Zip code:" are run together. 
They appear to be on the same line, confusing my screen reader as to 
which label to use for the Address line 2 field.
  * Related to the above, the label for Zip code appears to be missing 
when I tab to the field.
  * Definitely shorten the Bio legend, as tabbing to that field is a bit 
chatty.
  * I'm not sure what the Copy button near the custom URL field does. 
Maybe that could be made a bit clearer?
  * It might be useful to indicate how many degrees a photo is rotated 
so that I, as a screen reader user, know the state of my upload, if I 
accidentally clicked the button, etc.

**Comment by joshdanielson** [comment]
@ndarilek thank you! 

* the zip code was missing the id for the label, fixing that now. 
* I'm going to upload some revised, nested legends to see if that helps a bit 
* I'll add "Copy custom URL" to the copy button. Will that help?
* @IagoSRL I just created #639 for the rotate button fix. Good catch @ndarilek!

**Comment by joshdanielson** [comment]
@ndarilek I just committed my changes. I'm not sure how to make the legends less chatty as users have been confused, particularly because it's their contact information, e.g. who can see this? Any ideas to reduce it are welcome.

**Comment by IagoSRL** [comment]
@ndarilek I did progress on make 'tabs' used at the about and terms activities accessible; there is no an specific issue so requesting here if you can review them; is at the is528 branch, thanks.

**Comment by ndarilek** [comment]
Clicking the Terms of Service page doesn't speak a page title on load. 
Is this because there are no headers?

It might be useful to mark this page up with headers, or some other 
semantic elements that would make skimming easier.

The About activity is nice. It is well marked-up, and clicking the 
various tab links on top sets focus to and speaks the respective 
heading. Excellent!

**Comment by joshdanielson** [comment]
@ndarilek can you try again the terms: http://localhost:8811/appDebug.html#!terms/terms-of-service?

I cleaned up the html and added:
- section, header, ul, and li tags

**Comment by ndarilek** [comment]
Sorry, I'm not seeing any differences. I've checked both master and 
is528 branches. Is it somewhere else?

Note that, as far as headers go, you're better served by <h1/>, <h2/>, 
etc. I'm not sure of any screen readers that semantically analyze 
<header/> tags in sections to make them easily navigable, but <h1>-<h6> 
can generally be reached via the 1-6 keys for quicker skimming. 
<header/> is good for larger layouts, but <h1/>-<h6/> is better for 
longer document navigation.

**Comment by joshdanielson** [comment]
@ndarilek yes, it is in the 528 branch. And I mean h tags, not header and used h1, h2, and h3 tags. Can you try again with this link: http://localhost:8811/appDebug.html#!terms

**Comment by ndarilek** [comment]
When I visit that link at commit 63cdc7f, the markup looks like:

                 <p> <strong>1. Introduction</strong>

                 </p>
                 <p>Welcome to Loconomics.</p>
                 <p>These Terms of Service (“Agreement”) contain the 
terms on which Loconomics
...

I'd expect:

<h1>Introduction</h1>
...

Am I on the wrong commit? `git status` says I'm in is528.

Thanks.

**Comment by joshdanielson** [comment]
@ndarilek maybe you didn't sync the latest commits? I usually have to do multiple commits. I did most of the work in this commit: https://github.com/loconomics/loconomics/commit/d11d64ba3c7687d09839a755c256154fcb59abde

**Comment by ndarilek** [comment]
Oh, this is interesting. I've checked out the exact commit you've 
reverenced, and have found the HTML for the ToS in the repository. It is 
indeed formatted as you suggest.

I then use Firefox's Developer Tools, view the markup for the ToS in 
those, and it is still old-style as referenced in my earlier comment.

I just did rm -rf build/, and now I only have build/assets, no 
build/appDebug.html. Seems like maybe something didn't get updated even 
though I ran `grunt atwork`? How do I restore my build directory?

Apologies, if that's the case then my past 45 minutes of testing may 
have been invalid.

**Comment by ndarilek** [comment]
Oh, never mind, I just saw your docs about recreating /build. So sorry, 
been a long day. :)

**Comment by joshdanielson** [comment]
@ndarilek I've had some issues with this too and found that I kept using app.html instead of appDebug.html. I'm also using the Github desktop app on the mac so don't do anything other than grunt tasks in my terminal.

So, my suggestions are:
1. always sync all the latest commits in the branch
2. run "grunt build-dev atwork"
3. open a new tab with http://localhost:8811/appDebug.html

**Comment by ndarilek** [comment]
OK, we're good now. ToS looks great! Apologies for the confusion, I 
thought `atwork` was enough to rebuild everything.

**Comment by joshdanielson** [comment]
@ndarilek great! No worries at all. And did you check the other policies? You should be able to tab through them in the page navigation.

**Comment by ndarilek** [comment]
Yup, checked the privacy policy as well, it looked OK. Tab navigation 
works as you've described it should.

**Comment by joshdanielson** [comment]
Closing and keeping some sub issues open.
